controller/share_manager_controller.go

```diff
@@ -394,11 +394,13 @@ func (c *ShareManagerController) isShareManagerRequiredForVolume(volume *longhor
 	}
 
 	for _, attachmentTicket := range va.Spec.AttachmentTickets {
-		if isCSIAttacherTicketOfRegularRWXVolume(attachmentTicket, volume) {
-			return true
+		if isRegularRWXVolume(volume) && (isCSIAttacherTicket(attachmentTicket) || isUpgraderTicket(attachmentTicket)) {
+			// no active workload, there is no need to keep the share manager around
+			if hasActiveWorkload(volume) {
```

Comment:
I thought that the existence of `isCSIAttacherTicket(attachmentTicket) || isUpgraderTicket(attachmentTicket)` already indicate that `hasActiveWorkload(volume)` is true? If so, do we need the nested:
```
if hasActiveWorkload(volume) {
    return true
}
```

Reply:
ping @derekbit

Reply:
Yeah, the check is redundant. Remove it https://github.com/longhorn/longhorn-manager/pull/1993. @PhanLe1010  Can you help review it? Thank you.